File: 240201/CHJmission020101.c
#include <stdio.h>
#include <wiringPi.h>

#define PIR_PIN 2  // PIR 센서 연결된 GPIO 핀 번호
#define BUZZER_PIN 1  // 부저 연결된 GPIO 핀 번호

int main(void) {
    int pir_val;
    int buzzer_val;

    if (wiringPiSetup() == -1)
        return -1;

    pinMode(PIR_PIN, INPUT);
    pinMode(BUZZER_PIN, OUTPUT);

    printf("PIR Sensor and Buzzer Test\n");

    while (1) {
	pir_val = digitalRead(PIR_PIN);
        if (pir_val ==1) {
	    buzzer_val = 1;
            printf("Motion detected! Buzzing...\n");
            digitalWrite(BUZZER_PIN, buzzer_val);  // 부저 울리기
            delay(1000);  // 1초 동안 부저 울리기
	}
	else { 
	    buzzer_val = 0;
	    printf("Buzzer off\n");
            digitalWrite(BUZZER_PIN, buzzer_val);   // 부저 끄기
            delay(500);  // 0.5초 대기
        }

        delay(100);  // 0.1초 대기
	}
    return 0;
}

File: 240201/CHJmission020101.py
import RPi.GPIO as GPIO
import time

PIR_PIN = 27  # PIR 센서 연결된 GPIO 핀 번호
BUZZER_PIN = 18  # 부저 연결된 GPIO 핀 번호

GPIO.setmode(GPIO.BCM)
GPIO.setup(PIR_PIN, GPIO.IN)
GPIO.setup(BUZZER_PIN, GPIO.OUT)

print("PIR Sensor and Buzzer Test")

try:
    while True:
        motion_detected = GPIO.input(PIR_PIN)

        if motion_detected:
            print("Motion detected! Buzzing...")
            GPIO.output(BUZZER_PIN, GPIO.HIGH)  # 부저 울리기
            time.sleep(1)  # 1초 동안 부저 울리기
            GPIO.output(BUZZER_PIN, GPIO.LOW)   # 부저 끄기
            time.sleep(0.5)  # 0.5초 대기

        time.sleep(0.1)  # 0.1초 대기

except KeyboardInterrupt:
    pass

finally:
    GPIO.cleanup()

